src/typescript/frontend/src/app/chats/route.ts

```diff
@@ -0,0 +1,43 @@
+import { fetchChatEvents } from "@/queries/market";
+import type { NextRequest } from "next/server";
+import { stringifyJSON } from "utils";
+
+type ChatSearchParams = {
+  marketID: string | null;
+  fromMarketNonce: string | null;
+};
+
+export type ValidChatSearchParams = {
+  marketID: string;
+  fromMarketNonce: string;
+};
+
+const isNumber = (s: string) => !isNaN(parseInt(s));
+
+const isValidChatSearchParams = (params: ChatSearchParams): params is ValidChatSearchParams => {
+  const { marketID, fromMarketNonce } = params;
+  // prettier-ignore
+  return (
+    marketID !== null && isNumber(marketID) &&
+    fromMarketNonce !== null && isNumber(fromMarketNonce)
+  );
+};
+
+export async function GET(request: NextRequest) {
```

Comment:
I realize after trying to help Greg with our /api routes, we should probably add a simple example as a jsdoc string above these

You can probably write one up really quick with ChatGPT or something, should be pretty straightforward, just ask it to write a jsdoc but give it the context on the search params and any caveats, etc, then tell it to write an example